### ISSUE TITLE
Login gives wrong location

### ISSUE THREAD

**Issue by makitso** [fisharebest/webtrees#3522]
I have two trees on my site.  I would like to validate a user to be an editor on one tree, johannsen, and a member peters on the other.
When I log in with this userid I get to My Page for the wrong tree -- peters.

It would be nice if I could restrict a user to a specific, johannsen, tree when they log in.

**Comment by makitso** [comment]
Would be nice if you could set the default tree on a user basis.

**Comment by fisharebest** [comment]
When you log in, it is normal to stay on the same page (but with new access permissions).

In user-interface design, this is often called the [principle of least surprise](https://en.wikipedia.org/wiki/Principle_of_least_astonishment]).

So, if you are looking at an individual on tree1 and then login (perhaps to see more information or to create an edit), you would expect to be returned to the same page - not to a completely different page in a different tree.

**Comment by makitso** [comment]
> So, if you are looking at an individual on tree1 and then login (perhaps to see more information or to create an edit), you would expect to be returned to the same page - not to a completely different page in a different tree.

That makes sense for an experienced user.  However, if a user just wants to look at a specific tree, the non-default one, they have to set the tree on the home page prior to logging in -- something my casual users fail to do.

**Comment by DaPoHou** [comment]
It is also recommended to add the corresponding relationship between the tree and the user.
There are trees A and B. If a is the administrator of tree A but not the administrator of tree B, then a can only manage users of tree A in the user management in the control panel.

**Comment by DaPoHou** [comment]
> > So, if you are looking at an individual on tree1 and then login (perhaps to see more information or to create an edit), you would expect to be returned to the same page - not to a completely different page in a different tree.
> 
> That makes sense for an experienced user. However, if a user just wants to look at a specific tree, the non-default one, they have to set the tree on the home page prior to logging in -- something my casual users fail to do.

Yes, in all product designs, we must consider the ease of operation for new users.
If the corresponding relationship between different domain names and trees can be established, then the user accesses the corresponding default tree through the corresponding domain name. This may be more convenient.

**Comment by jsloane** [comment]
What about the case where you have 1 public tree used to demonstrate the system, and multiple private trees that users work on. It's currently not possible to navigate to a private tree, before logging on.
Where confidential data is kept private, users would expect to see private data after logging in and see their private tree that they're interested in.
Maybe a new "default tree" option on the user account might work. If this option is not set, the current workflow would remain.
